desk/app/expose.hoon

```diff
@@ -0,0 +1,2233 @@
+::  exposé: clearweb content rendering
+::
+::    copy a reference to a notebook post, then:
+::    :expose [%show /that/reference/with/id/numbers/quoted/'123456789']
+::    then visit in the browser:
+::    /expose/that/reference/as/copied/123456789
+::
+/-  c=cite, d=channels, co=contacts
+/+  sigil, u=channel-utils,
+    dbug, verb
+::
+|%
++$  state-0
+  $:  %0
+      open=(set cite:c)  ::TODO  could support ranges of msgs?
+  ==
+::
++$  action
+  $%  [%show =path]
+      [%hide =path]
+  ==
+::
++$  card  card:agent:gall
+::
+++  e
+  |%
+  ++  render-post
+    |=  [our=@p now=@da]
+    |=  [=nest:g:c msg=post:d]
+    ^-  (unit manx)
+    =/  aco=(unit contact:co)
+      =/  base=path  /(scot %p our)/contacts/(scot %da now)
+      ?.  .^(? %gu (weld base /$))
+        ~
+      =+  .^(rol=rolodex:co %gx (weld base /all/contact-rolodex))
+      ?~  for=(~(get by rol) author.msg)
+        ~
+      ?.  ?=([[@ ^] *] u.for)
+        ~
+      `con.for.u.for
+    ::
+    ::TODO  if we render replies then we can "unroll" whole chat threads too (:
```

Comment:
this would be really useful imo, worth doing

Reply:
My hesitation here is that it publicized other people's content, which you don't have control over, and other people may not expect.

(And also, keeping the replies up-to-date would require additional machinery. Though, realizing now, we currently don't reflect edits either...)

Reply:
yeah I noticed these are snapshots. do we disallow exposing posts from others?

Reply:
We don't disallow it, but probably won't expose that in the UI.

I should just make exposé listen to the channels firehose and do a refresh/re-render if one of the posts it cares about is affected.